File: 프로그래머스/2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.py
def solution(s):
    answer = []
    start = False
    
    for c in s:
        if c == ' ':
            start = False
        elif start:
            c = c.lower()
        else:
            c = c.upper()
            start = True
        
        answer.append(c)
        
    return ''.join(answer)

File: 프로그래머스/2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.java
public class Solution {
    
    public String solution(String s) {
        StringBuilder answer = new StringBuilder();
        boolean isFirst = true;

        for (char c : s.toCharArray()) {
            if (c == ' ') {
                answer.append(c);
                isFirst = true;
                continue;
            }
            if (isFirst) {
                answer.append(Character.toUpperCase(c));
                isFirst = false;
            } else {
                answer.append(Character.toLowerCase(c));
            }
        }

        return answer.toString();
    }
}